cluster-autoscaler/dynamic/config.go

```diff
@@ -0,0 +1,86 @@
+package dynamic
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/golang/glog"
+	"k8s.io/contrib/cluster-autoscaler/nodegroup"
+	"k8s.io/kubernetes/pkg/api/v1"
+)
+
+// Config which represents not static but dynamic configuration of cluster-autoscaler which would be updated periodically at runtime
+type Config struct {
+	Settings
+	resourceVersion string
+}
+
+// Settings of cluster-autoscaler contained in the latest config, which should be consumed by cluster-autoscaler
+type Settings struct {
```

Comment:
That's because I'd wanted to differentiate settings and their version. Settings are loaded from  a JSON object associated to the `settings` key in a configmap while a version is loaded from the metadata of it.